mantidproject/mantid
# Add in sample geometry columns

**Description of work.**

This adds the sample geometry settings to the table in the SANS GUI rather than the settings tab. This is because they are very run dependent. In order not to flood the table with unnecessary columns they are hidden by default.

**Report to:** steve 

**To test:**
 * Code review
 * Get the SANS data attached to this PR #23664
 * Make sure the data is added to your directories path and then load MaskFile.txt as the user file and batch_mode_reduction.csv as the batch file.
 * Check that the thickness column updates correctly 
 * Check that when you select the show geometry box at the bottom the additional geometry columns pop up and are correctly filled.



Partially Fixes #23209  




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
